### ISSUE TITLE
Narrator can't read the fabric Combobox options when expanded

### ISSUE THREAD

**Issue by sergul** [OfficeDev/office-ui-fabric-react#10689]
### Environment Information

- **Package version(s)**: (7.37.0)
- **Browser and OS versions**: (Microsoft Edge, Windows 10)

> Note that our screen reader support is limited to Edge + Narrator. Issues that do not reproduce in Edge + Narrator, and aren't caused by obvious invalid aria values, should be filed with the respective screen reading software, not the Fabric repo.

### Describe the issue:

Narrator is unable to read the Fabric combobox component options when expanded on arrow down key press.

#### Actual behavior:

Narrator is not able to read the combobox options, when expanded, by pressing the arrow down key press

#### Expected behavior:

Narrator should be able to read the combobox options, when expanded, by pressing the arrow down key press

### Documentation describing expected behavior

![combobox issue](https://user-images.githubusercontent.com/2910631/66076168-0fdfad80-e512-11e9-8aee-be8d0b8ad3c8.gif)

@micahgodbolt

**Comment by micahgodbolt** [comment]
i've reprod this. The issue is when the combo box opens, narrator doesn't say which item it is on, or what item number it is. The lack of repeating that "option A" is selected isn't super bad....but not know that it is X of 10 means it's hard to know where to go from there.

**Comment by micahgodbolt** [comment]
@jspurlin is this expected functionality? Do we need to trigger a selection when the combo box opens so that narrator things something has changed, and reads it out?

**Comment by jspurlin** [comment]
@micahgodbolt it looks like narrator is in scan mode, @sergul can you confirm? If so we've had this conversation a few time the issue is a narrator issue. Looping in @aneeshack4 and @dzearing who were in the earlier conversations around this. I believe a bug has already been logged against the Narrator team for this

**Comment by micahgodbolt** [comment]
I'm not sure if it's a scan issue. The question is, when the combobox opens, does it read the selection and position of the default selected item in the list?

Moving up or down 1 will read "option B, 2 of 10", but you don't get that selection/location data upon opening the combobox.

**Comment by micahgodbolt** [comment]
If you open the Dropdown, you will get "Banana, 2 of 7" as soon as the dropdown opens:

https://developer.microsoft.com/en-us/fabric#/controls/web/dropdown

**Comment by jspurlin** [comment]
How does a native html select element work? We should (at least) be inline with whatever that does.

For combobox we have aria-activedescendant set for the item that is "focused" (and make the selected item the focused item on open). From there it's on Narrator to announce whatever it wants.

For dropdown, it looks like we move focus to the "selected" item and again narrator will announce whatever it wants

**Comment by micahgodbolt** [comment]
yeah, it's the focus that doesn't get set. It gets set as soon as you arrow up or down to the next element, but you don't get focus on the selected item, so you don't know where you are on the list.

**Comment by micahgodbolt** [comment]
Narrator focuses puts focus on the selected item (in my case option 6) and reads out the option and location

> option 6, combo box, collapsed, Alt+ Down Arrow, 
option 6, <say-as interpret-as="spell">6</say-as> of <say-as interpret-as="spell">9</say-as>, selected,

**Comment by micahgodbolt** [comment]
it's the freeform situations that are difficult. Does focus move to the dropdown, and read out your current location? or do you keep focus in the input for future typing

**Comment by micahgodbolt** [comment]
@sergul so it might come down to narrator properly supporting [aria-activedescendant](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_aria-activedescendant_attribute)

**Comment by sergul** [comment]
> @micahgodbolt from the gif it looks like narrator is in scan mode, @sergul can you confirm? If so we've had this conversation a few time the issue is a narrator issue. Looping in @aneeshack4 and @dzearing who were in the earlier conversations around this. I believe a bug has already been logged against the Narrator team for this

**Comment by sergul** [comment]
> @micahgodbolt from the gif it looks like narrator is in scan mode, @sergul can you confirm? If so we've had this conversation a few time the issue is a narrator issue. Looping in @aneeshack4 and @dzearing who were in the earlier conversations around this. I believe a bug has already been logged against the Narrator team for this


@jspurlin narrator wasn't in scan mode and it was in tab mode.

**Comment by sergul** [comment]
> @sergul so it might come down to narrator properly supporting [aria-activedescendant](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_aria-activedescendant_attribute)

thanks @micahgodbolt 
My observation here though is if you open the combobox by left click then it will work and we will see the expected behavior for this issue. Do you think we should see different behaviors only because we open the combobox in different ways ? 
![combobox issue](https://user-images.githubusercontent.com/2910631/66084813-67871480-e524-11e9-9376-0c63f31fa30a.gif)

**Comment by jspurlin** [comment]
@micahgodbolt aria-activedescendant exists to solve exactly these kind of problems (having focus stay in the input put be able to navigate through the menu options). I know narrator has read the option that is being navigated (and says x of y) in the fabric comboBoxes, so unless that's regressed it should still work. I guess it's not clear to me exactly what the issue is... If it's a regression in narrator then we would open a bug on them

This component is marked up correctly and we shouldn't hack at it to try and get a different readout, that's the wrong approach.

**Comment by micahgodbolt** [comment]
@jspurlin i'm good with this resolution. We can't move focus to the dropdown in this case (like we can the dropdown), so we are using aria-acivedescendant. It appears to me that narrator is not yet supporting this correctly (adding aria-acivedescendant does not announce the active position upon opening). 

Do we know where to open narrator bugs? I can get this added or check if it's there.

**Comment by khmakoto** [comment]
@micahgodbolt did you check for bugs on Narrator?

**Comment by micahgodbolt** [comment]
btw, went down the rabbit hole a bit and found an example linked to from w3c, and it seems we're consistent with their example:

http://oaa-accessibility.org/example/11/

going to file a bug

**Comment by micahgodbolt** [comment]
Bug filed with narrator, linked back to this issue.

**Comment by khmakoto** [comment]
@sergul let us know if there is anything else we can do for you here. Otherwise, since we've filed a bug with Narrator, feel free to close this issue or let the bot close it for you. Thanks!
